### PR TITLE
Move to golang 1.12.1 official image

### PR DESCRIPTION
**Pull request by dims**
Remove the hack we introduced and switch jobs to newer image

### CODE REVIEW

**Comment by dims** [comment]
cc @tpepper @spiffxp

**Comment by dims** [comment]
related to https://github.com/kubernetes/kubernetes/issues/75372

File: images/kubekins-test/Dockerfile-1.14 [hunk 1]
```diff
@@ -17,51 +17,6 @@
 
 FROM golang:1.12
```
Comment (amwat) [inline images/kubekins-test/Dockerfile-1.14]:
```suggestion
FROM golang:1.12.1
```

Comment (dims) [reply images/kubekins-test/Dockerfile-1.14]: Whoops. fixed.

**Comment by spiffxp** [comment]
/approve
/lgtm
/hold
for @amwat to deploy when ready

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml [hunk 2]
```diff
@@ -13,6 +13,6 @@ periodics:
     path_alias: "sigs.k8s.io/cluster-api-provider-aws"
   spec:
     containers:
-    - image: gcr.io/k8s-testimages/kubekins-e2e:v20190315-49d62eb51-1.12
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20190316-dae90a14f-1.12
       command:
-      - "./scripts/ci-aws-cred-test.sh"
```
Comment (imkin) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: Is this an intentional change?

Comment (spiffxp) [reply config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: addition of a newline from `sed` or smth?

**Comment by amwat** [comment]
FWIW I don't think we need to bump anything since kubekins-e2e already used go1.12.1 from the release tarballs instead of basing on the golang image. We just the need the kubekins-test change which is only used currently by kubernetes-integration test (? @krzyzacy ).

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml (hunk 2 above)
Comment (imkin) [reply config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: Seems like a behavior of BSD "sed" on Mac. 
https://stackoverflow.com/questions/13325138/why-does-sed-add-a-new-line-in-osx
So the answer to my question is its unintentional and benign change. Infact we might appreciate the addition of the "newline" char

**Comment by dims** [comment]
@krzyzacy @amwat please feel free to open a fresh PR with the first commit or just edit this PR to drop the second commit if the changes are not needed. (i just rebased to master)

**Comment by amwat** [comment]
Updated the PR and pushed the new image. 
/lgtm
/hold

@krzyzacy @dims for final merging.

**Comment by dims** [comment]
/hold cancel

thanks @amwat !

**Pull request merged**